core/server/master/src/main/java/alluxio/master/file/meta/UfsSyncPathCache.java

```diff
@@ -80,18 +83,26 @@ public void notifySyncedPath(String path, DescendantType descendantType) {
    * @return true if a sync should occur for the path and interval setting, false otherwise
    */
   public boolean shouldSyncPath(String path, long intervalMs, boolean isGetFileInfo) {
+    SyncTime lastSync = mCache.getIfPresent(path);
+    if (lastSync == null) {
+      LOG.trace("Sync record not found, trigger sync");
+      return true;
+    }
```

Comment:
This is just a hack so our PoC can work in our manual reproduction. Obviously there's a problem here. If the user specifies syncInterval=-1 then I think we should still give the user the option to force not to sync. If we disregard syncInterval=-1 and just decide to sync when there's no record, the master will sometimes sync unnecessarily. For example a file is written through Alluxio, or a file is not sync-ed but the user just wants to disable sync whatsoever.

I think there's still a need to distinguish a forced sync (fix incorrect length) vs a never sync-ed file.